### PR TITLE
Fix concealing network errors in auth.whoami() and auth.isLoggedIn()

### PR DESCRIPTION
**Pull request by thgreasi**
Resolves: #805
Change-type: patch
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

---
##### Contributor checklist

- [x] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: lib/auth.coffee [hunk 1]
```diff
@@ -98,8 +98,8 @@ getAuth = (deps, opts) ->
 	###
 	exports.whoami = (callback) ->
 		getUserDetails()
-		.then (userDetails) -> userDetails.username || undefined
-		.catchReturn(undefined)
+		.then (userDetails) -> userDetails?.username || undefined
+		.catchReturn(new errors.BalenaNotLoggedIn, undefined)
```
Comment (pdcastro) [inline lib/auth.coffee]: Is the `new` keyword misplaced here? (There's no `new` in line 234 below.)

Comment (thgreasi) [reply lib/auth.coffee]: Nice catch @pdcastro :+1:

File: tests/integration/auth.spec.coffee [hunk 2]
```diff
@@ -128,6 +128,51 @@ describe 'SDK authentication', ->
...
+
+		describe 'balena.auth.isLoggedIn()', ->
+
+			it 'should eventually be false', ->
+				promise = balena.auth.isLoggedIn()
+				m.chai.expect(promise).to.eventually.be.false
+
+		describe 'balena.auth.whoami()', ->
+
+			it 'should eventually be false', ->
```
Comment (josecoelho) [inline tests/integration/auth.spec.coffee]: Quick typo: `it 'should eventually be undefined', ->`

Comment (thgreasi) [reply tests/integration/auth.spec.coffee]: Nice catch @josecoelho :+1:

Comment (thgreasi) [reply tests/integration/auth.spec.coffee]: @joehounsham here is a PR for this
See: https://github.com/balena-io/balena-sdk/pull/825

**Pull request merged**